src/collection-view.js

```diff
@@ -101,7 +101,7 @@ Marionette.CollectionView = Marionette.View.extend({
   _onCollectionAdd: function(child, collection, opts) {
     var index;
     if (opts.at !== undefined) {
-      index = opts.at;
+      index = opts.index !== undefined ? opts.index : collection.indexOf(child);
```

Comment:
Strange, but spec for filtered collection passed with my current code.. May be I'm wrong in spec?